common/include/pcl/point_representation.h

```diff
@@ -70,17 +70,15 @@ namespace pcl
        *  the point was reinterpret_casted to a float array of length nr_dimensions_ . This value says that this
        *  representation can be trivial; it is only trivial if setRescaleValues() has not been set.
        */
-      bool trivial_;
+      bool trivial_ = false;
 
     public:
       using Ptr = boost::shared_ptr<PointRepresentation<PointT> >;
       using ConstPtr = boost::shared_ptr<const PointRepresentation<PointT> >;
 
-      /** \brief Empty constructor */
-      PointRepresentation () : nr_dimensions_ (0), alpha_ (0), trivial_ (false) {}
-
       /** \brief Empty destructor */
-      virtual ~PointRepresentation () {}
+      virtual ~PointRepresentation () = default;
+      //TODO: check if copy and move constructors / assignment operators are needed
```

Comment:
At a first glance I would say no. But I don't have a complete view of the entire code base

Reply:
There are a few pitfalls in just declaring them defaults, specially in exception unsafe code. It might be better not to have them declared than to declare buggy defaults. Maybe a work for another PR

Reply:
Please expand, why defaults are buggy? And also, how implicit declared destructor (i.e. destructor created by the compiler when user does not declare one) is any different by the default one?
What pitfalls are you referring to when code is not exception safe?

Reply:
> how implicit declared destructor  ...  is any different by the default one?

That's not what I was referring to. Your edit is perfect. I was commenting on my own statement of "what's the issue with blind rule of 5".

> What pitfalls are you referring to when code is not exception safe?

Essentially, during copy, if there is an error, the rollback should result in the same state as before attempted copy. But with a) potentially self-copy b) copy of structs with resource owning semantics c) other similar corner cases, the rule of 5 might not be a good idea to apply blindly because the defaults do a element-wise operations. There might be scenarios where based on usage, we might have to do a copy into local variable before moving the resources and local variables to ensure that if copy fails, rollback is trivial.

Reply:
I see your point - b), a) not really - nonetheless I would say that this might happen when your class has non-trivially copyable elements, but in the case of PODs or classes that hold ``std::string``s or ``std::vector<int>``s, probably rule of 5 ( or even 7 ) is just fine.
I know even ``std::string`` copy constructor might fail with ``std::bad_alloc``, but again, running out of memory it's like an atomic bomb dropped in front of your house: who cares if the windows were closed or not (i.e. destructors are called or not ) .

Reply:
We are meandering out of context of the PR. But this is as good place as any.
> running out of memory it's like an atomic bomb dropped in front of your house

`bad_alloc` isn't called just due to OOM. It's also called when 
1. a single allocation is larger than the one allowed by the allocator (eg: differing implementations of malloc).  <-- most common reason
* allocation takes too much time and it'll impact the "real-time"ness of the application. In soft real-time systems, page swaps required for a large allocation can take time which results in a throw (all custom malloc support maximum time to wait)  <-- common in robotics
* application has run out of it's quota of memory (not the OS). This approach is frequently used in sandboxes and extra memory is given to application to clean itself or to recover (via custom api calls)  <-- common in cloud based virtualization
* there's a custom memory management (FreeRTOS, distributed memory, etc.) which can throw `bad_alloc` for other reasons too  <-- common in virtualization, hypervisors

Running out of memory is a very rare scenario for `bad_alloc`. Case 1 (common case) is essentially atom bomb for common users because it's not recoverable (by design or bad-calculations leading to the point) but not for all users.